tests/model_serving/model_server/kserve/canary_rollout/utils.py

```
@@ -0,0 +1,306 @@
+    """Assert Route traffic split by HTTP status fingerprint (200=stable, 500=canary)."""
+    inference = Inference(inference_service=isvc)
+    host = inference.get_inference_url()
+    predict_model = model_name or isvc.name
+    rest_url = f"https://{host}{V2_INFER_PATH_TEMPLATE.format(model_name=predict_model)}"
+
+    stable_hits = 0
+    canary_hits = 0
+    other_statuses: dict[int, int] = {}
+
+    for _ in range(sample_size):
+        response = requests.post(url=rest_url, json=TRAFFIC_INFERENCE_INPUT, verify=False, timeout=60)
```

Ruff (0.15.21)

[error] Probable use of `requests` call with `verify=False` disabling SSL certificate checks

(S501)